src/highlights/grid_line.ts

```diff
@@ -0,0 +1,187 @@
+import { cloneDeep, findLast } from "lodash";
+
+import { Highlight, HighlightRange, LineCell, NormalHighlightRange, VimCell, VirtualHighlightRange } from "./types";
+import { CellIter, getWidth, isDouble, splitGraphemes } from "./util";
+
+/**
+ * A class to manage the grid lines of a grid.
+ * Handle the grid_line event and be responsible for computing the highlight ranges.
+ */
+export class GridLine {
```

Comment:
just a naming thing: I think this should be plural (I thought this was only managing one line at first :) )

Reply:
GridLine does indeed cause confusion; typically, one would think it only manages a single row. However, using the plural form also feels awkward to me. Nevertheless, it doesn't matter much; there are comments, and no chance for reuse.

Reply:
Could also do something like GridLineTracker, but I won't push on this too hard

Reply:
Maybe `GridLineHandler`?

Reply:
Something like that sure. I was trying to emphasize that it stores data, but I could live with Handler